gs.js

```diff
@@ -122,9 +127,65 @@ class GlobalStorage extends lib.StorageProvider {
   ) {
     return this.nextId++;
   }
+
+  // Get system suffix for given id
+  //   id - <common.Uint64>
+  //
+  // Returns: <common.Uint64>
+  getSystemSuffix(id) {
+    return common.Uint64.and(id, this.systemBitmask);
+  }
+
+  // Check whether data with given id is stored on this system
+  //   id - <common.Uint64>
+  //
+  // Returns: <boolean>
+  curSystem(id) {
+    return common.Uint64.cmp(this.getSystemSuffix(id), this.systemSuffix);
+  }
+
+  // Get server suffix for given id
+  //   id - <common.Uint64>
+  //
+  // Returns: <common.Uint64>
+  getServerSuffix(id) {
+    return common.Uint64.and(id, this.serverBitmask);
+  }
+
+  // Check whether data with given id is stored on this server
+  //   id - <common.Uint64>
+  //
+  // Returns: <boolean>
+  curServer(id) {
+    return common.Uint64.cmp(this.getServerSuffix(id), this.serverSuffix);
+  }
+
+  // Get id without system and server suffix
+  //   id - <common.Uint64>
+  //
+  // Returns: <common.Uint64>
+  getLocalId(id) {
+    const bitmask = common.Uint64.or(this.serverBitmask, this.systemBitmask);
```

Comment:
I think you can skip this step now since `serverBitmask` will most likely include the `systemBitmask` in it. So it is going to be just
```js
  return common.Uint64.and(common.Uint64.not(this.serverBitmask), id);
```